relations/jf/src/deposit.rs

```diff
@@ -1,85 +1,67 @@
+        Self {
+            deposit_note: SourcedNote {
+                note: public.note,
+                token_id: public.token_id,
+                token_amount: public.token_amount,
+                trapdoor: private.trapdoor,
+                nullifier: private.nullifier,
+                note_type: NoteType::Deposit,
+            },
+        }
+    }
+}
+
+#[derive(Default)]
 pub struct DepositPublicInput {
     pub note: Note,
     pub token_id: TokenId,
     pub token_amount: TokenAmount,
 }
 
-impl Marshall for DepositPublicInput {
-    fn marshall(&self) -> Vec<CircuitField> {
-        vec![
-            convert_hash(self.note),
-            self.token_id.into(),
-            self.token_amount.into(),
-        ]
+impl PublicInput for DepositRelation {
+    fn public_input(&self) -> Vec<CircuitField> {
+        self.deposit_note.public_input()
     }
 }
 
-#[derive(Copy, Clone, Eq, PartialEq, Hash, Default, Debug)]
+#[derive(Default)]
 pub struct DepositPrivateInput {
     pub trapdoor: Trapdoor,
     pub nullifier: Nullifier,
 }
 
 impl Relation for DepositRelation {
-    type PublicInput = DepositPublicInput;
-    type PrivateInput = DepositPrivateInput;
-
-    fn new(public_input: Self::PublicInput, private_input: Self::PrivateInput) -> Self {
-        Self {
-            public: public_input,
-            private: private_input,
-        }
-    }
-
     fn generate_subcircuit(&self, circuit: &mut PlonkCircuit<CircuitField>) -> PlonkResult<()> {
-        // Register public inputs.
-        let note_var = circuit.create_public_variable(convert_hash(self.public.note))?;
-        let token_id_var = circuit.create_public_variable(self.public.token_id.into())?;
-        let token_amount_var = circuit.create_public_variable(self.public.token_amount.into())?;
-
-        // Register private inputs.
-        let trapdoor_var = circuit.create_variable(convert_hash(self.private.trapdoor))?;
-        let nullifier_var = circuit.create_variable(convert_hash(self.private.nullifier))?;
-
-        // Ensure that the token amount is valid.
-        // todo: extract token amount limiting to at least constant, or even better to a function/type
-        circuit.enforce_leq_constant(token_amount_var, CircuitField::from(u128::MAX))?;
-
-        let zero_var = circuit.create_constant_variable(CircuitField::zero())?;
-
-        // Check that the note is valid.
-        // todo: move to a common place
-        let inputs: [usize; 6] = [
-            token_id_var,
-            token_amount_var,
-            trapdoor_var,
-            nullifier_var,
-            zero_var,
-            zero_var,
-        ];
-        let computed_note_var = RescueNativeGadget::<CircuitField>::rescue_sponge_no_padding(
-            circuit,
-            inputs.as_slice(),
-            1,
-        )?[0];
-
-        circuit.enforce_equal(note_var, computed_note_var)?;
+        let note = SourcedNote {
```

Comment:
not needed, right? we can use directly `self.deposit_note`

Reply:
oh yes, this is from previous version, forgot to change that.